doc/reference/storage/fcb/fcb.rst

```diff
@@ -0,0 +1,77 @@
+.. _fcb:
+
+Flash Circular Buffer (FCB)
+###########################
+
+Flash circular buffer provides an abstration through which you can treat
+flash like a FIFO. You append entries to the end, and read data from the
+beginning.
+
+Description
+***********
+
+Elements in the flash contain the length of the element, the data within
+the element, and checksum over the element contents.
+
+Storage of elements in flash is done in a FIFO fashion. When user
```

Comment:
To be consistent, I'd recommend using "you" instead of "user" in this section, so ``When you request space...`` and ``When you start reading...`` and ``You have control...`` in the next paragraph.  (Search for ``user``)